pkg/utils/utils.go

```diff
@@ -1195,3 +1195,36 @@ func CompareVersion(currentVersion string, minVersion string) int {
 	}
 	return current.Compare(minimum)
 }
+
+func RetrieveVddkLogLevel(params map[string]interface{}, config *rest.Config, logger logrus.FieldLogger) error {
+	var err error // Declare here to avoid shadowing on using in cluster config only
+	if config == nil {
+		config, err = GetKubeClientConfig()
+		if err != nil {
+			logger.WithError(err).Errorf("Failed to get k8s inClusterConfig")
+			return errors.Wrap(err, "Could not retrieve in-cluster config")
+		}
+	}
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logger.WithError(err).Errorf("Failed to get k8s clientset from the given config: %v", config)
+		return err
+	}
```

Comment:
use the util, CreateKubeClientSet, to replace L1201 ~ L1212.

BTW, you don't need to have `config` parameter in the input since you always call the function by setting it to nil.